application/ui/components/params.jsx

```diff
@@ -0,0 +1,189 @@
+                    break;
+            }
+        });
+
+        return {
+            values      : values,
+            nullKeys    : {},
+            includeKeys : includes
+        };
+    },
+
+    getValue : function()
+    {
+        var data = {}, paramName;
+
+        for (var i in this.props.params) {
+            paramName = this.props.params[i].name;
+            if (typeof this.state.includeKeys[paramName] !== 'undefined') {
+                if (typeof this.state.nullKeys[paramName] === 'undefined') {
+                    data[paramName] = this.state.values[paramName];
+                } else {
+                    data[paramName] = null;
+                }
+            }
+        }
+
+        return data;
+    },
+
+    /**
+     * Manage null values
+     * @param key
+     * @param value
+     */
+    nullHandler : function(key, value)
+    {
+        var state = _.extend({}, this.state);
+
+        if (value) {
+            state.nullKeys[key] = true;
+        } else {
+            delete state.nullKeys[key];
+        }
+
+        this.props.onChange(this.getValue());
+        this.setState(state);
+    },
+
+    includeHandler : function(key, value)
+    {
+        var state = _.extend({}, this.state);
+
+        if (value) {
+            state.includeKeys[key] = true;
+        } else {
+            delete state.includeKeys[key];
+        }
+
+        this.setState(state);
+        this.props.onChange(this.getValue());
+    },
+
+    changeHandler : function(key, value)
+    {
+        var state = _.extend({}, this.state);
+
+        state.values[key] = value;
+
+        this.setState(state, function() {
+            this.props.onChange(this.getValue());
+        });
+    },
+
+    // ensures values are passed up even if none have changed
+    componentDidMount: function()
+    {
+        this.props.onChange(this.getValue());
+    },
+
+    renderParams : function()
+    {
+        var rows = [];
+        var component = this;
+
+        _.each(this.props.params, function(item) {
+            rows.push(
+                <ParamRow
+                    key         = {item.name}
+                    param       = {item}
+                    value       = {component.state.values[item.name]}
+                    onChange    = {_.partial(component.changeHandler, item.name)}
+                    onInclude   = {_.partial(component.includeHandler, item.name)}
+                    onNull      = {_.partial(component.nullHandler, item.name)}
+                    isNull      = {typeof component.state.nullKeys[item.name] !== 'undefined'}
+                    isIncluded  = {typeof component.state.includeKeys[item.name] !== 'undefined'}
+                    method      = {component.props.endpointMethod}
+                />
+            );
+        });
+
+        return rows;
+    },
+
+    render : function()
+    {
+        if (this.props.params.length === 0) {
+            return (
+                <span />
+            );
```

Comment:
You can just `return;` here (or `return null;`) since the desired effect is "don't render anything".

React explicitly understands `null`/`undefined` to mean "there's actually nothing to render here."